src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -323,6 +328,11 @@ void BaseCouplingScheme::secondExchange()
 
   if (reachedEndOfTimeWindow()) {
 
+    // If preCICE has stopped before the end of the time window we have to duplicate the last available sample and put it at the end of the time window.
+    if (getNextTimeStepMaxSize() > math::NUMERICAL_ZERO_DIFFERENCE && hasTimeWindowSize()) {
+      addTimeStepAtWindowEnd();
+    }
```

Comment:
```suggestion
```
I don't think you need to do this twice. Doing this once (in `firstExchange` or `secondExchange`) should be sufficient.

Reply:
Yes thanks, I had misunderstood what first and second exchange do.